Infrastructure/Componants/Product.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Collections.Generic;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Interactions;
+using Core;
+
+namespace Infrastructure
+{
+    public class Product : ComponantBase
+    {
+        private IWebElement AddToCartButton =>  ParentElement.WaitAndFindElement(By.CssSelector(".button.ajax_add_to_cart_button.btn.btn-default span"));
```

Comment:
Check if you don't have to contain all the classes, maybe try with only .add_to_cart_button?
Also, you can use By.ClassName()